src/net_processing.cpp

```diff
@@ -146,6 +146,8 @@ static constexpr uint32_t MAX_GETCFILTERS_SIZE = 1000;
 static constexpr uint32_t MAX_GETCFHEADERS_SIZE = 2000;
 /** the maximum percentage of addresses from our addrman to return in response to a getaddr message. */
 static constexpr size_t MAX_PCT_ADDR_TO_SEND = 23;
+/** The compactblocks version we support. See BIP 152. */
+static constexpr uint64_t CMPCTBLOCKS_VERSION = 2;
```

Comment:
The version field needs to be 8 bytes (uint64_t). If we change this const to a uint8_t, then the message serialized here:

```
            m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, /* high_bandwidth= */ false, /* version= */ CMPCTBLOCKS_VERSION));
```

would have a payload size of 2 (1 byte for high_bandwidth, and 1 byte for version). That would fail to be parsed by other clients.

Reply:
That's a bit brittle in any case. Perhaps put `uint64_t{CMPCTBLOCKS_VERSION}` there instead?

Reply:
"followed by a 8-byte integer" -> I misread, thanks

Reply:
or perhaps:

```
            static_assert(sizeof(CMPCTBLOCKS_VERSION) == sizeof(uint64_t), "sendcmpct version field must be 8 bytes");
            m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, /* high_bandwidth= */ false, /* version= */ CMPCTBLOCKS_VERSION));
```

(why check at runtime what you can test at compile time?)

Reply:
uint64_t{CMPCTBLOCKS_VERSION}` is a compile time check ;)

Reply:
I've added the static_assert.

Reply:
> uint64_t{CMPCTBLOCKS_VERSION} is a compile time check ;)

`uint64_t{CMPCTBLOCKS_VERSION}` initializes a new `uint64_t` from `CMPCTBLOCKS_VERSION` at runtime, no? But I guess the compiler will optimize that away?

In any case, I've removed the static assert and replaced with `uint64_t{CMPCTBLOCKS_VERSION}`.